File: EcommerceChatbot/Models/DetailBestellung.cs
using EcommerceChatbot.Models;
using System.Text.Json.Serialization;

public class DetailBestellung
{
    public int Id { get; set; }
    public int BestellungId { get; set; }
    public int ProduktId { get; set; }
    public int Menge { get; set; }
    public decimal Preis { get; set; }
    
    // Propriétés de navigation comme nullable
    [JsonIgnore]
    public Bestellung? Bestellung { get; set; }
    
    [JsonIgnore]
    public Product? Produkt { get; set; }
}

File: ecommerce-frontend/src/models/DetailBestellung.ts
import { Bestellung } from "./Bestellung";
interface DetailBestellung {
  id?: number;
  produktId: number;
  menge: number;
  preis: number;
  produkt?: {
    name: string;
    price: number;
  };
  bestellungId?: number;
  bestellung?: Bestellung;
}
export { DetailBestellung };
